pageserver/src/tenant/storage_layer/image_layer.rs

```diff
@@ -661,6 +666,17 @@ impl ImageLayerInner {
             };
         }
     }
+
+    async fn iter<'a, 'ctx>(&'a self, ctx: &'ctx RequestContext) -> ImageLayerIterator<'a, 'ctx> {
```

Comment:
Why is this async, do you really need both lifetimes?

Reply:
removed async; I encountered some weird lifetime errors when removing the lifetimes. Probably there is a way to fix that, but I prefer having two lifetimes to make it clear when using and passing these fields.